docs/understanding-dependencies.md

```diff
@@ -0,0 +1,24 @@
+# dependencies vs. devDependencies
+
+> What's the difference between these two properties in `package.json`?
```

Comment:
I wonder if, even before this, it'd be helpful to add a bit of info about `package.json`. Something like:

> Modern web development projects, whether created by Guppy or through the command line, use Node Package Manager (NPM) to manage dependencies (for example, React is a dependency because it's an external package that your code relies on).
>
> Each project that uses NPM has a manifest called `package.json` in the main directory of the project. An example `package.json` looks like this:
>
> // Basic package.json with a couple dependencies and devDependencies